packages/core/core/src/RequestTracker.js

```diff
@@ -222,24 +222,25 @@ export class RequestGraph extends ContentGraph<
 
   // addNode for RequestGraph should not override the value if added multiple times
   addNode(node: RequestGraphNode): NodeId {
-    let didNodeExist = this.hasContentKey(node.id);
+    let nodeId = this._contentKeyToNodeId.get(node.id);
```

Comment:
```suggestion
    let nodeId = this.getNodeIdByContentKey(node.id);
```

Reply:
Except it throws if not there 😢

Reply:
Should we change it to return null instead and do the throwing in the caller if needed?

Reply:
I think there was some reason that @thebriando made it throw. But `_contentKeyToNodeId` is somewhat private

Reply:
When we wrote it, we wanted the caller to be sure that if you were `getting` a node id by content key, that it should exist. 

Could be: `if(this.hasContentKey(node.id))` then assign nodeId after the conditional.

Reply:
Yeah the problem I was trying to solve was the multiple hash table lookups in this function since it is pretty hot.

Reply:
Well in `Graph.js`, it's also "get" but doesn't throw: `getNode(id: NodeId): ?TNode`

Reply:
So either we leave it as is, or add a non-throwing version of `getNodeIdByContentKey`.

Changing `getNodeIdByContentKey` would cause 179 Flow errors.

Reply:
I vote to leave it as is for now.